WordPressAuthenticator/Unified Auth/View Related/Get Started/GetStartedViewController.swift

```diff
@@ -516,16 +518,21 @@ private extension GetStartedViewController {
     /// Show the Password entry view.
     ///
     func showPasswordView() {
-        guard let vc = PasswordViewController.instantiate(from: .password) else {
-            DDLogError("Failed to navigate to PasswordViewController from GetStartedViewController")
+        guard let navigationController = navigationController else {
             return
         }
-
-        vc.source = source
-        vc.loginFields = loginFields
-        vc.trackAsPasswordChallenge = false
-
-        navigationController?.pushViewController(vc, animated: true)
+        configureViewLoading(true)
+        let coordinator = PasswordCoordinator(navigationController: navigationController,
+                                              source: source,
+                                              loginFields: loginFields,
+                                              tracker: tracker,
+                                              configuration: WordPressAuthenticator.shared.configuration)
+        passwordCoordinator = coordinator
```

Comment:
❓ Why are we storing this in `passwordCoordinator`? 

I tried removing this and noticed that we need this [here](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/74830ef88500460a525ad4b0f103b0b1e25a44d1/WordPressAuthenticator/Unified%20Auth/View%20Related/Password/PasswordCoordinator.swift#L72)

- Instead of asking `PasswordCoordinator` to present the password view, can we pass the necessary properties to `MagicLinkRequestedViewController` and ask it present the screen?
- We could even create a separate coordinator just for showing `PasswordViewController` and store it in `MagicLinkRequestedViewController`

I am sure that you considered the above and several other options, and I am curious to know your thoughts. 😄